src/server/replica.cc

```diff
@@ -367,7 +367,7 @@ error_code Replica::InitiatePSync() {
     io::PrefixSource ps{io_buf.InputBuffer(), Sock()};
 
     // Set LOADING state.
-    CHECK(service_.SwitchState(GlobalState::ACTIVE, GlobalState::LOADING) == GlobalState::LOADING);
+    CHECK(service_.SwitchState(GlobalState::ACTIVE, GlobalState::LOADING).second);
```

Comment:
I don't remember in depth.... When we start replication, we immediately set the global state in `ReplicaOfInternal` and we don't switch back to active. So here, we don't expect it always to be active, we just try to make sure its in loading (either already or by switch)